io/FV3GFS_io.F90

```diff
@@ -2638,101 +2638,122 @@ subroutine fv3gfs_diag_output(time, diag, atm_block, nx, ny, levs, ntcw, ntoz, &
            endif
          endif
          if (diag(idx)%axes == 2) then
-           if (trim(diag(idx)%mask) == 'positive_flux') then
-             !--- albedos are actually a ratio of two radiation surface properties
-             var2(1:nx,1:ny) = 0._kind_phys
-             do j = 1, ny
-               jj = j + jsc -1
-               do i = 1, nx
-                 ii = i + isc -1
-                 nb = Atm_block%blkno(ii,jj)
-                 ix = Atm_block%ixp(ii,jj)
-                 if (Diag(idx)%data(nb)%var21(ix) > 0._kind_phys) &
-                   var2(i,j) = max(0._kind_phys,min(1._kind_phys,Diag(idx)%data(nb)%var2(ix)/Diag(idx)%data(nb)%var21(ix)))*lcnvfac
+           ! Integer data
+           int_or_real: if (associated(Diag(idx)%data(1)%int2)) then
+             if (trim(Diag(idx)%intpl_method) == 'nearest_stod') then
+               var2(1:nx,1:ny) = 0._kind_phys
+               do j = 1, ny
+                 jj = j + jsc -1
+                 do i = 1, nx
+                   ii = i + isc -1
+                   nb = Atm_block%blkno(ii,jj)
+                   ix = Atm_block%ixp(ii,jj)
+                   var2(i,j) = real(Diag(idx)%data(nb)%int2(ix), kind=kind_phys)
+                 enddo
                enddo
-             enddo
-           elseif (trim(Diag(idx)%mask) == 'land_ice_only') then
-             !--- need to "mask" gflux to output valid data over land/ice only
-             var2(1:nx,1:ny) = missing_value
-             do j = 1, ny
-               jj = j + jsc -1
-               do i = 1, nx
-                 ii = i + isc -1
-                 nb = Atm_block%blkno(ii,jj)
-                 ix = Atm_block%ixp(ii,jj)
-                  if (Diag(idx)%data(nb)%var21(ix) /= 0) var2(i,j) = Diag(idx)%data(nb)%var2(ix)*lcnvfac
+               call store_data(Diag(idx)%id, var2, Time, idx, Diag(idx)%intpl_method, Diag(idx)%name)
```

Comment:
This GitHub diff view is confusing. The only thing that changed was adding the `int_or_real: if` case and move the existing code into the `else` block.

Reply:
Maybe in the future we need to consider output those fields as integer too.

Reply:
Yes, that would be nice, but it will require changes to the pre- and post-processing chain.